src/widgets/index/index.ts

```diff
@@ -357,6 +361,7 @@ const index = (props: IndexProps): Index => {
       // listener on `change` below, once `init` is done.
       helper.on('change', ({ isPageReset }) => {
         if (isPageReset) {
+          localUiState.page = undefined;
```

Comment:
I would have put this condition where we mutate the `localUiState` that way it's simpler to reason about.